lib/arel/middleware/database_executor.rb

```diff
@@ -46,7 +48,12 @@ def call(next_arel)
 
       def execute_sql(next_arel)
         sql, binds = next_arel.to_sql_and_binds
+
+        # TODO: don't set cache if the binds != context[:original_binds]
```

Comment:
I have to verify but I think we should not cache if the bind params are reordered.